brian2genn/templates/max_row_length_array.cpp

```diff
@@ -0,0 +1,58 @@
+{# USES_VARIABLES { _synaptic_pre, _synaptic_post, sources, targets,
+                    N_incoming, N_outgoing, N,
+                    N_pre, N_post, _source_offset, _target_offset }
+#}
+{# WRITES_TO_READ_ONLY_VARIABLES { N_incoming, N_outgoing}
+#}
+
+#include "brianlib/common_math.h"
+#include "brianlib/stdint_compat.h"
+#include<cmath>
+#include<ctime>
+#include<iostream>
+#include<fstream>
+#include<climits>
+#include "brianlib/stdint_compat.h"
+#include "synapses_classes.h"
+
+////// SUPPORT CODE ///////
+namespace {{codeobj_name}}_array {
+	{{support_code_lines|autoindent}}
+}
+
+////// HASH DEFINES ///////
+{{hashdefine_lines|autoindent}}
+
+void _run_{{codeobj_name}}()
+{
+    using namespace brian;
+    using namespace {{codeobj_name}}_array;
+
+    ///// CONSTANTS ///////////
+    %CONSTANTS%
+
+    ///// POINTERS ////////////
+    {{pointers_lines|autoindent}}
+  
+    const size_t _old_num_synapses = {{N}};
+    const size_t _new_num_synapses = _old_num_synapses + _numsources;
```

Comment:
What are these used for and how come you're adding numbers of neurons to synapses?

Reply:
Ah - not used here - related to the original synapses_create_array template. I will remove.

Reply:
Actually ... rather leave them there as they *might* relate to some of the python generate code lines (I have not inspected this in detail but note that these variables are also not used explicitly in the original template)

Reply:
I didn't really have time to follow this in detail today but just to clear up some potential confusion: the `sources` and `targets` in this template correspond to arrays of source indices and target indices that should be connected (i.e. the arguments in `synapses.connect(i=[...], j=[...])`. The `_numsources`/``_numtargets` are the lengths of these arrays and therefore the number of *synapses* that will be added by this template.

All that said, I think both variables are left-over from some earlier version of the template and indeed neither used here nor in the original synapse creation template. Feel free to delete them.

Reply:
Ahh I see - thanks @mstimberg

Reply:
right ... so I will remove them in the next commit.